packages/enclave-contracts/contracts/Enclave.sol

```diff
@@ -243,7 +247,7 @@ contract Enclave is IEnclave, OwnableUpgradeable {
     function publishInput(
         uint256 e3Id,
         bytes calldata data
-    ) external returns (bool success) {
+    ) external nonReentrant returns (bool success) {
```

Comment:
I'd probably refrain from using the guard here as well. What are the risks? let's imagine a request for an e3 is accepted (let's remember that all of these parameters are public at the time that a e3 request is made, even before a committee is selected) with a input validator contract that performs some reentrancy to the Enclave contract, what could they be doing? 

The contract re-enters and submits another input with different data, the data would be validated and inserted in the tree. That is no different than calling the publishInput function twice by two different addresses. 

Another option is that they re-enter with the same data, then `inputs[e3Id]._insert(inputHash);` will fail because you cannot insert two equal leaves into a LeanIMT. 

Anything I'm missing?